src/pybind/mgr/dashboard/frontend/cypress/e2e/page-helper.po.ts

```diff
@@ -13,17 +13,19 @@ export abstract class PageHelper {
    */
   static restrictTo(page: string): Function {
     return (target: any, propertyKey: string, descriptor: PropertyDescriptor) => {
-      const fn: Function = descriptor.value;
-      descriptor.value = function (...args: any) {
-        cy.location('hash').should((url) => {
-          expect(url).to.eq(
-            page,
-            `Method ${target.constructor.name}::${propertyKey} is supposed to be ` +
-              `run on path "${page}", but was run on URL "${url}"`
-          );
-        });
-        fn.apply(this, args);
-      };
+      if (descriptor) {
+        const fn: Function = descriptor?.value;
+        descriptor.value = function (...args: any) {
```

Comment:
types

Reply:
i actually don't know the type here.

the actual interface where this is being passed is defined with `any` type
```
/**
     * Calls the function, substituting the specified object for the this value of the function, and the specified array for the arguments of the function.
     * @param thisArg The object to be used as the this object.
     * @param argArray A set of arguments to be passed to the function.
     */
    apply(this: Function, thisArg: any, argArray?: any): any;
```
and `arg` is passed to `thisArg`. I could say something like `string[]` but not sure if that could cause any regressive behavior.